h/static/scripts/controllers/share-widget-controller.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+const Controller = require('../base/controller');
+
+const TEMPLATE_SELECTOR = '.share-widget-template';
+const CONFIG_ATTR = 'share-widget-config';
+const TRIGGER_SELECTOR = `[${CONFIG_ATTR}]`;
+const WIDGET_SELECTOR = '.share-widget';
+const TARGET_HREF_ATTR = 'share-target-href';
+const TARGET_HREF_SELECTOR = `[${TARGET_HREF_ATTR}]`;
+const CLIPBOARD_INPUT_SELECTOR = 'input.share-widget-link__text';
+const PRIVATE_MSG_SELECTOR = '.share-widget-msg-private';
+const GROUP_MSG_SELECTOR = '.share-widget-msg-group';
+
+const ARROW_PADDING_RIGHT =  16;
+const ARROW_PADDING_BOTTOM =  5;
+
+
+// Travel up the nodes parents to find a matching
+// element with the share-widget configuration values
+const _getParentWithConfig = function(node) {
```

Comment:
We should be consistent about using arrow functions here. This ought to have been flagged up by ESLint, will have to look into why that did not happen.